conf.d/hydro.fish

```diff
@@ -61,11 +59,15 @@ function _hydro_prompt --on-event fish_prompt
     set --query _hydro_status || set --global _hydro_status "$_hydro_newline$_hydro_color_prompt$hydro_symbol_prompt"
     set --query _hydro_pwd || _hydro_pwd
 
-    command kill $_hydro_last_pid 2>/dev/null
+    set --local addons _hydro_addon_{$hydro_prompt_addons}\;
 
-    set --query _hydro_skip_git_prompt && set $_hydro_git && return
+    command kill $_hydro_last_pid 2>/dev/null
 
     fish --private --command "
```

Comment:
Because we are defining the prompt in private mode, it doesn't appear possible to conditionally display a character in the prompt if we are in private mode. Could we introduce an additional global variable (something like `is_fish_private_mode` that gets set just before we enter private mode here? Then we can have the following addon in the prompt:

```
set --global hydro_prompt_addons private

function _hydro_addon_private
  if test "$is_fish_private_mode" = 1
    echo [p]
  end
end
```

Reply:
We may not even need `--private` mode to be honest! 🤔

Reply:
I don't think we need to change anything in Hydro to handle this particular addon example. In your `config.fish`, you can just check that the shell is interactive so that subshells that run your config (like those Hydro creates) don't nuke your flag. This code will do what you want with the current implementation:

```fish
# config.fish
status --is-interactive && set --export is_fish_private_mode $fish_private_mode
function _hydro_addon_private
    if test "$is_fish_private_mode" = 1
        echo [p]
    end
end
```

I like having Hydro use `--private` so it doesn't clutter your fish history with all the non-user `git` commands it fires off, but that's not a huge deal either way. But, if this is work-around is acceptable, we can leave `--private` in and everyone wins.

Reply:
I concur!